lib/cortex/test_runner.ex

```diff
@@ -76,35 +81,64 @@ defmodule Cortex.TestRunner do
 
   def handle_call({:run_tests_for_file, path, _opts}, _from, state) do
     case test_file_for_path(path) do
-      :not_found ->
-        {:reply, :ok, state}
-
+      :not_found -> {:reply, :ok, state}
       test_path ->
-        files_to_load =
-          [test_helper(path), test_path]
-
-        compiler_errors =
-          files_to_load
-          |> Stream.map(&Reloader.reload_file/1)
-          |> Enum.reject(&(&1 == :ok))
+        case run_test_files([test_path]) do
+          :ok -> {:reply, :ok, state}
+          err = {:error, _} -> err
+        end
+    end
+  end
 
-        with [] <- compiler_errors do
-          task =
-            Task.async(ExUnit, :run, [])
+  def handle_call(:run_all, _from, state) do
+    case run_test_files() do
+      :ok -> {:reply, :ok, state}
+      err = {:error, _} -> err
+    end
+  end
 
-          ExUnit.Server.cases_loaded()
-          Task.await(task, :infinity)
+  def run_test_files, do: run_test_files(all_test_files())
+  def run_test_files([]), do: :ok
+  def run_test_files(files) do
+    files
+    |> Enum.group_by(&test_helper/1)
+    |> Enum.map(fn {helper, files} -> run_test_files(helper, files) end)
+    |> Enum.find(:ok, fn
+      :ok         -> false
+      {:error, _} -> true
+    end)
+  end
 
-          {:reply, :ok, state}
-        else
-          errors ->
-            compiler_error_descs =
-              errors
-              |> Stream.map(&elem(&1, 1))
-              |> Enum.join("\n")
+  def run_test_files(_test_helper, []), do: :ok
+  def run_test_files(test_helper, files) do
+    files_to_load = [test_helper | files]
+    compiler_errors =
+      files_to_load
+      |> Stream.map(&Reloader.reload_file/1)
+      |> Enum.reject(&(&1 == :ok))
+
+    with [] <- compiler_errors do
+      task = Task.async(ExUnit, :run, [])
+      ExUnit.Server.cases_loaded()
+      Task.await(task, :infinity)
+      :ok
+    else errors ->
+      compiler_error_descs =
+        errors
+        |> Stream.map(&elem(&1, 1))
+        |> Enum.join("\n")
+
+      {:error, compiler_error_descs}
+    end
+  end
 
-            {:reply, {:error, compiler_error_descs}, state}
-        end
+  defp all_test_files do
```

Comment:
This works fine for typical elixir apps, but as written won't grab any test files when cortex is run in an umbrella root – you'll need to collect all the test files across all the umbrella/apps/**, or some similar method. I poked around for a bit but am not sure the cleanest way to do it. Note `Mix.Project.umbrella?` for checking on that case.